src/write.c

```diff
@@ -567,15 +567,26 @@ static avifResult avifEncoderAddImageInternal(avifEncoder * encoder,
         return AVIF_RESULT_NO_CONTENT;
     }
 
-    if ((cellCount > 1) && ((firstCell->width < 64) || (firstCell->height < 64))) {
+    if ((cellCount > 1) && !avifAreGridDimensionsValid(firstCell->yuvFormat,
+                                                       gridCols * firstCell->width,
+                                                       gridRows * firstCell->height,
+                                                       firstCell->width,
+                                                       firstCell->height,
+                                                       &encoder->diag)) {
         return AVIF_RESULT_INVALID_IMAGE_GRID;
     }
 
     for (uint32_t cellIndex = 0; cellIndex < cellCount; ++cellIndex) {
         const avifImage * cellImage = cellImages[cellIndex];
-        if ((cellImage->depth != firstCell->depth) || (cellImage->width != firstCell->width) ||
-            (cellImage->height != firstCell->height) || (!!cellImage->alphaPlane != !!firstCell->alphaPlane) ||
-            (cellImage->alphaPremultiplied != firstCell->alphaPremultiplied)) {
+        // MIAF (ISO 23000-22:2019), Section 7.3.11.4.1:
```

Comment:
Please also cite the following requirement in HEIF Section 6.6.2.3.1:

> All input images shall have exactly the same width and height; call those `tile_width` and `tile_height`.

This requirement is enforced by the check `(cellImage->width != firstCell->width) || (cellImage->height != firstCell->height)`.

In our video call today, you mentioned that the right-most column or the bottom-most row can have a smaller tile width or tile height. Are you referring to this requirement in HEIF Section 6.6.2.3.1?

> The reconstructed image is formed by tiling the input images into a grid with a column width (potentially excluding the right-most column) equal to `tile_width` and a row height (potentially excluding the bottom-most row) equal to `tile_height`, without gap or overlap, and then trimming on the right and the bottom to the indicated `output_width` and `output_height`.

The "potentially excluding the right-most column" and "potentially excluding the bottom-most row" parts in this requirement seem to allow the right-most column to be narrower than `tile_width` and the bottom-most row to be shorter than `tile_height`. But it is not written clearly.